freebsd: various improvements and new packages

rhelmot opened the pull request:
## Description of changes

part of https://github.com/NixOS/nixpkgs/pull/296581

This is a big one, but all its individual pieces would be too small on their own for my taste; if you like I can break it up further. There are a lot of commits, but each one has a detailed message and usually only touches one file. I would recommend reviewing by opening all commits in new tabs and then going one at a time.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
  - [x] x86_64-freebsd
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

rhelmot committed:
freebsd.compat: re-add lost patch

This patch was applied to 14.0 but not 13.1. It is necessary for 13.1 as
well.

rhelmot committed:
freebsd.boot-install: wrap coreutils install instead of netbsd

netbsd can no longer compile under FreeBSD native early bootstrap
stdenv, so switch to coreutils. This only involves discarding the -l
flag. The -l flag causes a symlink instead of a copy to be installed, so
it is safe to discard during bootstrap.

rhelmot committed:
freebsd.filterSource: use a less featureful rsync

By adding fewer dependencies, the FreeBSD native bootstrap tarball
becomes lighter.

rhelmot committed:
freebsd.mkDerivation: process patches generated with git

The diff parsing was pretty hardcoded for diff -u. Now it should handle
git diff/show as well.

rhelmot committed:
freebsd.{libc,compat,include}: use no-libs stdenv

Previously, an attribute named isStatic did this, but this was lost in a
refactor. Revive it and rename it to noLibc to be more clear about its
intended use.

rhelmot committed:
freebsd.libxo: init

rhelmot committed:
freebsd.libelf: init

rhelmot committed:
freebsd.{libncurses,libncurses-tinfo}: init

These packages are preferred over libncurses proper for some in-tree
FreeBSD packages.

rhelmot committed:
freebsd.libedit: init

rhelmot committed:
freebsd.libjail: init

rhelmot committed:
freebsd.libdl: init

rhelmot committed:
freebsd.libcasper: init

rhelmot committed:
freebsd.libcapsicum: init

rhelmot committed:
freebsd.localedef: init

rhelmot committed:
freebsd.locales: init

The parameters to control which locales are built are placeholders.

rhelmot committed:
freebsd.locale: init

rhelmot committed:
freebsd.bin: init

This is a single derivation containing everything in /bin on a normal
FreeBSD machine, which is an incredibly compact set of core utilities.

rhelmot committed:
freebsd.cp: init

While cp is part of freebsd.bin, this particular package is critical for
early stdenv boot and has its own derivation to require fewer
dependencies

rhelmot committed:
freebsd.iconv: init

rhelmot committed:
freebsd.ldd: init

Ericson2314 commented:
Oh lol, I was just making this too. LGTM! Since this is isn't touching non-FreeBSD packages / existing things outside FreeBSD land so much, I don't care so much that it's big.

Ericson2314 commented on pkgs/os-specific/bsd/freebsd/pkgs/ldd.nix:
```diff
@@ -0,0 +1,21 @@
+  };
+
+  meta.platforms = lib.platforms.freebsd;
+})
```
I think the parens are unnedeed?

rhelmot replied:
Parens were added months ago as part of a mass rebuild avoidance hack that I just found and removed prior to this PR, but not formatting-wise!

Ericson2314 reviewed (approved):
Let's go!

Ericson2314 commented:
The eval failure looks legit

alyssais commented on pkgs/os-specific/bsd/freebsd/pkgs/boot-install.nix:
```diff
@@ -1,10 +1,20 @@
+    fixed_args=()
+    while [[ ''${#args[0]} > 0 ]]; do
+      case "''${args[0]}" in
+        -l)
```
Would be nice to explain in comment or commit message what -l does and why it's safe to ignore.

rhelmot replied:
Done

rhelmot replied:
Fixed!

alyssais commented on pkgs/os-specific/bsd/freebsd/pkgs/libncurses.nix:
```diff
@@ -0,0 +1,25 @@
+  ...
+}:
+mkDerivation {
+  path = "lib/ncurses/ncurses";
```
Preferred why?

rhelmot replied:
`/nix/store/wf6yhi6c33dv5n18vj68pj8cim7k4bqa-binutils/bin/ld: cannot find -ltermcapw: No such file or directory`

freebsd's ncurses provides different libraries from normal ncurses.

alyssais commented on pkgs/os-specific/bsd/freebsd/pkgs/libedit.nix:
```diff
@@ -0,0 +1,7 @@
+{ mkDerivation, libncurses-tinfo }:
+mkDerivation {
+  path = "lib/libedit";
```
Is this different from all the libedit variants we have packaged at the top level?

rhelmot replied:
Yes. Various packages depend on filecomplete.h, which is not provided by the standard libedit. It _is_ provided by netbsd.libedit, but I would rather not mix freebsd and netbsd libs.

alyssais commented on pkgs/os-specific/bsd/freebsd/pkgs/libcasper.nix:
```diff
@@ -0,0 +1,26 @@
+  libnv,
+}:
+mkDerivation {
+  path = "lib/libcasper/libcasper";
```
We removed libcasper before, and now we're adding it back?

rhelmot replied:
You're referring to the earlier diff in this PR? that's for the compat generation, which is separate from the actual package (this) which is a dependency of various things. Compat is just for building packages for the build machine.

rhelmot replied:
I updated the commit message for the compat patch indicating this.

alyssais commented on pkgs/os-specific/bsd/freebsd/patches/14.0/localedef.patch:
```diff
@@ -0,0 +1,209 @@
++	{ "IS2",		'\x1e' },
++	{ "US",			'\x1f' },
++	{ "IS1",		'\x1f' },
++	{ "DEL",		'\x7f' },
```
What's going on with all these additions?

rhelmot replied:
@artemist this is your patch

rhelmot replied:
Based on my investigation, this doesn't actually impact the building of locales as we currently have it configured, and only adjusts the default charmap, which is used if you don't specify any charmap file at generation time. Diff indicates that there's no actual difference between the locales derivation with or without this patch, so I'm removing it.

alyssais commented on pkgs/os-specific/bsd/freebsd/pkgs/locales.nix:
```diff
@@ -0,0 +1,51 @@
+let
+  build =
+    name: needsLocaledef:
+    mkDerivation {
```
Do we really need to use 11 derivations for this, rather than 1?

rhelmot replied:
This is without a doubt the most elegant way to tackle this build using the infrastructure we have set up for building packages out of the FreeBSD source tree. Each derivation is invoking a separate makefile in a separate directory and has a separate set of build inputs to ensure no cross-contamination.

The alternative would be basically overriding buildPhase and installPhase to build and install with several makefiles.

What is the downside of many small derivations, even if they're not user-facing?

Ericson2314 replied:
I indeed want to have a `libc0` so we can stop doing all the extra libs in `postInstall` for that too (future work I am planning on).

alyssais commented on pkgs/os-specific/bsd/freebsd/pkgs/bin.nix:
```diff
@@ -0,0 +1,92 @@
+    }mkbuiltins mksyntax mktokens mknodes
+    make -C $BSDSRCDIR/bin/csh $makeFlags "CC=${pkgsBuildBuild.stdenv.cc}/bin/cc" CFLAGS="-D__unused= -D__printf0like\(a,b\)= -D__dead2= -I$BSDSRCDIR/contrib/tcsh -I." ${
+      lib.optionalString (!stdenv.buildPlatform.isFreeBSD) "MK_PIE=no "
+    }gethost
```
Why are other utilities independently packaged but these are not?

rhelmot replied:
build fails unless these cflags are supplied, but only for these utilities. They end up being packaged together, they just can't be built with the same code.

alyssais commented on pkgs/os-specific/bsd/freebsd/pkgs/bin.nix:
```diff
@@ -0,0 +1,92 @@
+  gencat,
+}:
+mkDerivation {
+  pname = "bins";
```
Why do we need a single derivation with every binary that FreeBSD has?  It seems weird that we need both sh and csh, for example, when every build environment will already have bash.  Also that groff and mandoc are included, when anywhere else they'd be separately specified inputs.

alyssais commented on pkgs/os-specific/bsd/freebsd/pkgs/iconv.nix:
```diff
@@ -0,0 +1,12 @@
+  libcasper,
+}:
+mkDerivation {
+  path = "usr.bin/iconv";
```
libiconv in all-packages.nix should be updated to use this.  Is that coming later?

rhelmot replied:
https://github.com/NixOS/nixpkgs/pull/311789/files#diff-ab5748dc9567516fefba8344056b51ec1866adeace380f46e58a7af3d619ea22R22488

rhelmot commented on pkgs/os-specific/bsd/freebsd/patches/13.1/compat-install-dirs.patch:
```diff
@@ -2,12 +2,11 @@ diff --git a/tools/build/Makefile b/tools/build/Makefile
  # yet (this happens if we are crossbuilding from Linux/Mac).
  INSTALLDIR_LIST= \
 -	bin \
--	lib/casper \
```
Looks like I misremembered where this patch came from. This patch actually dates back to 0afe9d1f70434c02b1efb84629b75385a57936cf, and is just being added back now after an accidental deletion. I'll update the commit message.

rhelmot committed:
freebsd.libcxxrt: fixup in preparation for inclusion in stdenv

- add option to build without libcxx
- don't build with libcxx
- distribute headers

paparodeo commented:
the `freebsd.libcxxrt` closure seems like it contains the whole llvm toolchain

<details><summary>nix-store --query --tree $(nix-build -A pkgsCross.x86_64-freebsd.freebsd.libcxxrt)</summary>

```
/nix/store/9a0k2rrpyiwxbgqb3a0v1gmvs0z7n95c-libcxxrt-x86_64-unknown-freebsd-14.0.0
├───/nix/store/9lzbpwsl4wqg7x10mdni6gy120smlmmy-libc-x86_64-unknown-freebsd-14.0.0
│   └───/nix/store/9lzbpwsl4wqg7x10mdni6gy120smlmmy-libc-x86_64-unknown-freebsd-14.0.0 [...]
└───/nix/store/90i7wmzlkv7rfgnd0ij2d78sy3s14bki-x86_64-unknown-freebsd-clang-wrapper-17.0.6
    ├───/nix/store/28gpmx3z6ss3znd7fhmrzmvk3x5lnfbk-gnugrep-3.11
    │   ├───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52
    │   │   ├───/nix/store/1q9vc0lq7qjlfjz47mfmlzdf86c543jy-xgcc-13.2.0-libgcc
    │   │   ├───/nix/store/ic63ay0py10fyryaw7345k4ps32da33w-libidn2-2.3.7
    │   │   │   ├───/nix/store/yfp7dr8m7zi7kxk49wd714gwvhb105hf-libunistring-1.1
    │   │   │   │   └───/nix/store/yfp7dr8m7zi7kxk49wd714gwvhb105hf-libunistring-1.1 [...]
    │   │   │   └───/nix/store/ic63ay0py10fyryaw7345k4ps32da33w-libidn2-2.3.7 [...]
    │   │   └───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   ├───/nix/store/g3vi60zgyjsvij7xkk6dxky1hkwh0ynd-pcre2-10.43
    │   │   ├───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   │   └───/nix/store/g3vi60zgyjsvij7xkk6dxky1hkwh0ynd-pcre2-10.43 [...]
    │   └───/nix/store/28gpmx3z6ss3znd7fhmrzmvk3x5lnfbk-gnugrep-3.11 [...]
    ├───/nix/store/306znyj77fv49kwnkpxmb0j2znqpa8bj-bash-5.2p26
    │   ├───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   └───/nix/store/306znyj77fv49kwnkpxmb0j2znqpa8bj-bash-5.2p26 [...]
    ├───/nix/store/30gv8ad6a58cjkhsg8f24id4bjdzpcsg-compiler-rt-x86_64-unknown-freebsd-17.0.6
    ├───/nix/store/9lzbpwsl4wqg7x10mdni6gy120smlmmy-libc-x86_64-unknown-freebsd-14.0.0 [...]
    ├───/nix/store/gnjp969f02l4fp2bj6vxibnb6c0qfqb5-expand-response-params
    │   └───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    ├───/nix/store/php4qidg2bxzmm79vpri025bqi0fa889-coreutils-9.5
    │   ├───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   ├───/nix/store/5nk2ga7i2f030am4qpcdsd8qlk6i3z83-attr-2.5.2
    │   │   ├───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   │   └───/nix/store/5nk2ga7i2f030am4qpcdsd8qlk6i3z83-attr-2.5.2 [...]
    │   ├───/nix/store/7ivacs3m2fm19hyxdmrs05xisj82v6y5-gmp-with-cxx-6.3.0
    │   │   ├───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   │   ├───/nix/store/xvzz97yk73hw03v5dhhz3j47ggwf1yq1-gcc-13.2.0-lib
    │   │   │   ├───/nix/store/0rxb3ixzk4zaqivc9s795m0a3679wbw2-gcc-13.2.0-libgcc
    │   │   │   ├───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   │   │   └───/nix/store/xvzz97yk73hw03v5dhhz3j47ggwf1yq1-gcc-13.2.0-lib [...]
    │   │   └───/nix/store/7ivacs3m2fm19hyxdmrs05xisj82v6y5-gmp-with-cxx-6.3.0 [...]
    │   ├───/nix/store/mjgi65m3hgzqqvqcix848gskkw5zzwi9-acl-2.3.2
    │   │   ├───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   │   ├───/nix/store/5nk2ga7i2f030am4qpcdsd8qlk6i3z83-attr-2.5.2 [...]
    │   │   └───/nix/store/mjgi65m3hgzqqvqcix848gskkw5zzwi9-acl-2.3.2 [...]
    │   └───/nix/store/php4qidg2bxzmm79vpri025bqi0fa889-coreutils-9.5 [...]
    ├───/nix/store/9mr86gvjgm254jmbw285f3vn1ic6z4iv-x86_64-unknown-freebsd-llvm-binutils-wrapper-17.0.6
    │   ├───/nix/store/306znyj77fv49kwnkpxmb0j2znqpa8bj-bash-5.2p26 [...]
    │   ├───/nix/store/9lzbpwsl4wqg7x10mdni6gy120smlmmy-libc-x86_64-unknown-freebsd-14.0.0 [...]
    │   ├───/nix/store/ar57w399dhm6rvc1v3zdnd1qdba04fc4-llvm-binutils-17.0.6
    │   │   ├───/nix/store/2a9v95z4v1jlmyzzpkiq78y4vf1b3nq6-llvm-17.0.6
    │   │   │   ├───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   │   │   ├───/nix/store/j4m3lwhh4x0bzilvi77d512ahry775nq-ncurses-6.4
    │   │   │   │   ├───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   │   │   │   └───/nix/store/j4m3lwhh4x0bzilvi77d512ahry775nq-ncurses-6.4 [...]
    │   │   │   ├───/nix/store/lv6nackqis28gg7l2ic43f6nk52hb39g-zlib-1.3.1
    │   │   │   │   └───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   │   │   ├───/nix/store/xvzz97yk73hw03v5dhhz3j47ggwf1yq1-gcc-13.2.0-lib [...]
    │   │   │   ├───/nix/store/3dw83mb8zpmfgdywlddmy6py661x9z9b-llvm-17.0.6-lib
    │   │   │   │   ├───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   │   │   │   ├───/nix/store/j4m3lwhh4x0bzilvi77d512ahry775nq-ncurses-6.4 [...]
    │   │   │   │   ├───/nix/store/kbg5m7fyi1w23fyfmxjhhzcbd577rpg0-libffi-3.4.6
    │   │   │   │   │   ├───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   │   │   │   │   └───/nix/store/kbg5m7fyi1w23fyfmxjhhzcbd577rpg0-libffi-3.4.6 [...]
    │   │   │   │   ├───/nix/store/lv6nackqis28gg7l2ic43f6nk52hb39g-zlib-1.3.1 [...]
    │   │   │   │   ├───/nix/store/kf7fzax0av9him9m4zcmbqmihh5108wq-libxml2-2.12.7
    │   │   │   │   │   ├───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   │   │   │   │   ├───/nix/store/lv6nackqis28gg7l2ic43f6nk52hb39g-zlib-1.3.1 [...]
    │   │   │   │   │   └───/nix/store/kf7fzax0av9him9m4zcmbqmihh5108wq-libxml2-2.12.7 [...]
    │   │   │   │   ├───/nix/store/xvzz97yk73hw03v5dhhz3j47ggwf1yq1-gcc-13.2.0-lib [...]
    │   │   │   │   └───/nix/store/3dw83mb8zpmfgdywlddmy6py661x9z9b-llvm-17.0.6-lib [...]
    │   │   │   └───/nix/store/5jd2n4igjfag33424vn8lyzwy3c6zvf7-libpfm-4.13.0
    │   │   │       └───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   │   └───/nix/store/5bmxfn5qdigwh019miq0yqaafjpvbywk-lld-17.0.6
    │   │       ├───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   │       ├───/nix/store/lv6nackqis28gg7l2ic43f6nk52hb39g-zlib-1.3.1 [...]
    │   │       ├───/nix/store/xvzz97yk73hw03v5dhhz3j47ggwf1yq1-gcc-13.2.0-lib [...]
    │   │       ├───/nix/store/3dw83mb8zpmfgdywlddmy6py661x9z9b-llvm-17.0.6-lib [...]
    │   │       └───/nix/store/zdz0gqraaqi0dczwcl6mn5vk3zfjjxl3-lld-17.0.6-lib
    │   ├───/nix/store/gnjp969f02l4fp2bj6vxibnb6c0qfqb5-expand-response-params [...]
    │   ├───/nix/store/php4qidg2bxzmm79vpri025bqi0fa889-coreutils-9.5 [...]
    │   └───/nix/store/9mr86gvjgm254jmbw285f3vn1ic6z4iv-x86_64-unknown-freebsd-llvm-binutils-wrapper-17.0.6 [...]
    ├───/nix/store/g65yzxk9vggn3s55svvl7370yqlgpblj-compiler-rt-x86_64-unknown-freebsd-17.0.6-dev
    │   └───/nix/store/30gv8ad6a58cjkhsg8f24id4bjdzpcsg-compiler-rt-x86_64-unknown-freebsd-17.0.6 [...]
    ├───/nix/store/xsrgnv6hbx1fhr8wmpaqpnh6vpkxxjy5-clang-17.0.6-lib
    │   ├───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   ├───/nix/store/xvzz97yk73hw03v5dhhz3j47ggwf1yq1-gcc-13.2.0-lib [...]
    │   ├───/nix/store/3dw83mb8zpmfgdywlddmy6py661x9z9b-llvm-17.0.6-lib [...]
    │   └───/nix/store/xsrgnv6hbx1fhr8wmpaqpnh6vpkxxjy5-clang-17.0.6-lib [...]
    ├───/nix/store/hwdd1q4h8vc6d27r1ppvq9xxg8k2d362-clang-17.0.6
    │   ├───/nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52 [...]
    │   ├───/nix/store/xvzz97yk73hw03v5dhhz3j47ggwf1yq1-gcc-13.2.0-lib [...]
    │   ├───/nix/store/3dw83mb8zpmfgdywlddmy6py661x9z9b-llvm-17.0.6-lib [...]
    │   └───/nix/store/xsrgnv6hbx1fhr8wmpaqpnh6vpkxxjy5-clang-17.0.6-lib [...]
    └───/nix/store/90i7wmzlkv7rfgnd0ij2d78sy3s14bki-x86_64-unknown-freebsd-clang-wrapper-17.0.6 [...]
```
</details>

rhelmot commented:
Looks like this problem traces to the debug symbols.

```
< 1><0x0000050b>    DW_TAG_typedef
                      DW_AT_type                  <0x000004eb>
                      DW_AT_name                  ptrdiff_t
                      DW_AT_decl_file             0x00000002 /nix/store/90i7wmzlkv7rfgnd0ij2d78sy3s14bki-x86_64-unknown-freebsd-clang-wrapper-17.0.6/resource-root/include/stddef.h
                      DW_AT_decl_line             0x00000023
```

This seems moderately intentional, so I'm not sure what to do about this besides patching, but this seems like it might affect other derivations as well.

alyssais commented:
separateDebugInfo?

paparodeo commented:
> separateDebugInfo?

it is already separate. i just added `outputs = [ "out" "dev" "debug" ];` to the freebsd.libcxxrt and everything worked. 

```
$ nix-store --query --tree $(nix-build -A pkgsCross.x86_64-freebsd.freebsd.libcxxrt)
/nix/store/ayhjr7cgvp5hx3b34dxm16rc21v8pp0q-libcxxrt-x86_64-unknown-freebsd-14.0.0
└───/nix/store/9lzbpwsl4wqg7x10mdni6gy120smlmmy-libc-x86_64-unknown-freebsd-14.0.0
    └───/nix/store/9lzbpwsl4wqg7x10mdni6gy120smlmmy-libc-x86_64-unknown-freebsd-14.0.0 [...]
```

Ericson2314 commented:
I hope we can get this merged soon, any final review thoughts @alyssais?

paparodeo commented on pkgs/os-specific/bsd/freebsd/pkgs/libcxxrt.nix:
```diff
@@ -8,4 +8,6 @@ mkDerivation {
   pname = "libcxxrt";
   path = "lib/libcxxrt";
   extraPaths = [ "contrib/libcxxrt" ];
+  noLibcxx = true;
```
```suggestion
  outputs = [ "out" "dev" "debug" ];
  noLibcxx = true;
```

rhelmot replied:
Applied this patch, tyvm!
